### PR TITLE
Added Extracting of URLs from blog feed function

### PR DESCRIPTION
**Pull request by smilegodly**
I wrote some code that is meant to get us started with extracting URL's from blog feeds . 
This code takes in a blog feed and gives back all the URL's in the feed.
All these URL's will later be categorized by what type of URL, is it a  pull request, a issue, etc..
Fixes: #53

### CODE REVIEW

**Comment by smilegodly** [comment]
I couldn't get some of the eslint errors (about indentation) to go away and might need some help with that in class later.

**Comment by manekenpix** [comment]
@smilegodly if you hover over the error it'll tell you what the problem is.
I managed to get rid of them:
Screenshot from 2019-11-08 00-45-54

Let me know if you need help with it.

File: src/extract-urls.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// The goal is to create a function that given a blog feed URL
+// it will extract the data from the webpage and parse the URLs from that feed
+// Later on , you can then organize the types of urls (github etc)...
+const bent = require('bent');
+
+const request = bent('string');
+
+const util = require('util');
+
+const spawn = require('child_process');
```
Comment (smilegodly) [inline src/extract-urls.js]: your change here killed the function...
```8 Nov 14:39:37 - spawning python process
(node:2599) UnhandledPromiseRejectionWarning: TypeError: spawn is not a function
    at request.then (/home/matty_smilegodly/gitRepos/telescope/src/extract-urls.js:18:21)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
(node:2599) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2599) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by smilegodly** [comment]
> This needs to be reworked. First, I don't want to take on a dependency on Python for this--there is nothing here that needs Python. We can, and should, do all of it in JS.

There are no extra dependencies for this as python3 is installed by default.. Also it states in the [overview](https://github.com/Seneca-CDOT/telescope/blob/master/docs/overview.md#an-initial-set-of-goals) in the MVP features that this particular MVP/Prototype feature can be written in both Python & Javascript...

**Comment by humphd** [comment]
@smilegodly python isn't installed by default on Windows.

At any rate, I'd ask that you work in JS, as there's nothing in this PR that requires something beyond JS.  If you needed something that couldn't be done in JS and node, that would be different.  But extracting URLs from text is not one of them.

**Comment by smilegodly** [comment]
> @smilegodly python isn't installed by default on Windows.
It is installed by default with Windows Subsystem for Linux ( WSL )
I dont see why I wouldn't use python as it is more readable and faster to develop in than nodejs.

**Comment by humphd** [comment]
@smilegodly happy to discuss with you in person on Monday.

**Comment by smilegodly** [comment]
@humphd @cindyledev I've made the requested changes, If there are any issues with this please let me know.

**Comment by smilegodly** [comment]
I have rebased.
Should be good to go.
@humphd

**Comment by humphd** [comment]
@smilegodly one more update, and I'll merge: the `feed-worker.js` has been changed to add parsing from XML.  You can now use the post Object, and get the html directly from it vs. using a network request.

**Comment by cindyledev** [comment]
Your package-lock and feed-worker have conflicts in them. Please address them

**Comment by smilegodly** [comment]
I have changed it to work with the post object. @humphd

**Comment by humphd** [comment]
Rebased and merged in https://github.com/Seneca-CDOT/telescope/commit/f59032790d5858442b21104e92139fe7b82827a5.

**Pull request closed without merging**